spec.html

```diff
@@ -14205,7 +14202,7 @@ <h1>Runtime Semantics: Evaluation</h1>
             1. Return *true*.
           1. If IsPropertyReference(_ref_) is *true*, then
             1. If IsSuperReference(_ref_) is *true*, throw a *ReferenceError* exception.
-            1. [id="step-delete-operator-toobject"] Let _baseObj_ be ! ToObject(_ref_.[[Base]]).
+            1. [id="step-delete-operator-toobject"] Let _baseObj_ be ? ToObject(_ref_.[[Base]]).
```

Comment:
This appears to change an edge case related to the `null` and `undefined` check--one that may be unintentional.

```js
class C {
  m() {
    delete super.x;
  }
}
Object.setPrototypeOf(C, null);
C.m();
```

In the spec today, evaluating `super.x` will produce a TypeError due to its use of RequireObjectCoercible (that is, before the runtime semantics of the `delete` operator begin).

With this change applied, evaluation makes it to the `delete` operator, but the ReferenceError from the IsSuperReference check will be thrown before ToObject has a chance to throw a TypeError.

There's implementation divergence here, too. Today, V8 and JSC throw the ReferenceError while SpiderMonkey and Engine262 throw the TypeError.

If my reading is accurate and we want to preserve the currently-specified behavior, it may be as simple as swapping this step and the one immediately before it.

Reply:
Pardon me: the method in the example code ought to be static. I'll update it accordingly.

Reply:
Thanks for pointing this out.

Personally I think the behavior in this PR is reasonable: checking whether something is a super reference seems more "static" than checking that the resolved value is object-coercible, so it makes sense to do the `super` check first. (In fact, I think it would have been possible to forbid `delete super.x` at parse time; I'm not sure why we made it a runtime error instead.)

Given that, and since we got consensus on this PR as written, I'm going to go ahead with the current behavior. But I'll call it out at the next meeting to make sure everyone is aware, and give people a chance to raise an issue if they'd like it modified.

Reply:
Got it. [Here's the corresponding change for Test262](https://github.com/tc39/test262/pull/3055).